### DISCUSSION TITLE
Cannot understand this NRT warning; maybe a bug?

### DISCUSSION THREAD

**Discussion by qrli** [dotnet/roslyn#73823] [Q&A] [answered]
The `exchangeName` variable should be non-null on line 28:
![image](https://github.com/dotnet/roslyn/assets/10579882/d856a4f7-5783-4038-bc92-d03fad96c2e0)

Visual Studio correctly says that `exchangeName` is not null:
![image](https://github.com/dotnet/roslyn/assets/10579882/6934480b-6ac4-48a6-8270-242b385b6d8a)

But the compiler issues warning saying it may be null:
![image](https://github.com/dotnet/roslyn/assets/10579882/f330156b-1781-4759-8e97-c7c4daf27131)

Why?

**Comment by CyrusNajmabadi** [comment]
can you supply a repro?

**Reply by qrli** [reply]
After try and error several times, I think I finally reduced it to minimum. It seems to be caused by `netstandard2.0`.
csproj:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFrameworks>net8.0;netstandard2.0</TargetFrameworks>
    <ImplicitUsings>enable</ImplicitUsings>
	  <LangVersion>10.0</LangVersion>
	  <Nullable>enable</Nullable>
  </PropertyGroup>

</Project>
```
Program.cs:
```cs
Console.WriteLine("Hello, World!");
new Foo().NRTIssue(null);

class Foo
{
    public void NRTIssue(string? s = null)
    {
        if (string.IsNullOrEmpty(s))
            s = "foo";
        M(s);
    }

    void M(string nonNullString)
    {
    }
}
```

**Reply by CyrusNajmabadi** [reply]
It is likely not annotated in that tfm.

**Reply by MaceWindu** [reply, marked as answer]
Yep, `IsNullOrEmpty` annotations appeared quite late, making multi-tfm projects being polluted by `!`'s

**Reply by qrli** [reply]
I came up with this solution:
```xml
  <PropertyGroup Condition="'$(TargetFramework)' == 'net6.0'">
    <nullable>Enable</nullable>
  </PropertyGroup>
  <PropertyGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
    <nullable>Annotations</nullable>
  </PropertyGroup>
```

**Reply by jasonmalinowski** [reply]
For what it's worth, @qrli's approach is what we do in Roslyn, effectively: only use the latest TFM for actual validation, the other ones annotation only.

**Reply by MaceWindu** [reply]
@qrli, that's a nice workaround. Unfortunatelly doesn't work well when you have tfm-specific code for which you need analysis.
